src/runtime/virtcontainers/utils/utils.go

```diff
@@ -390,3 +391,63 @@ outer:
 
 	return nil
 }
+
+func MkdirAllWithInheritedOwner(path string, perm os.FileMode) error {
```

Comment:
Can the utils additions be added as their own commit, and can we add unit tests?

Reply:
I previously added the unit tests but removed them because it requires root to run them (for chown). According to https://github.com/kata-containers/kata-containers/issues/2446, root is removed in the github action.

Reply:
@fengwang666 - understood, but that's a temporary bug. We do need tests for this function so you can add them and protect them using a simple approach like this:

- https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/mount_test.go#L307..L309

Or for more elaborate handling, you can use:

- https://github.com/kata-containers/kata-containers/tree/main/src/runtime/pkg/katatestutils

Reply:
Hey, root tests have been re-enabled by a PR merged earlier Today.